library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js

```diff
@@ -102,7 +114,12 @@ $.fn.folderListing = function(options) {
       }
 
       li.append(a);
-      html.append(li);
+      if(settings.displayRootDirectory && urlRoot === root){
```

Comment:
Then here just do `container.append(li)` and it doesn't matter how you got here.